### PR TITLE
Feature/allow subsets of properties

### PR DESCRIPTION
**Pull request by matt-akkio**
Allow us to pull subsets of properties for all of Contacts, Companies, and Deals

### CODE REVIEW

File: tap_hubspot/tap.py [hunk 1]
```diff
@@ -112,9 +112,15 @@ class TapHubspot(Tap):
...
             "start_date",
             th.DateTimeType,
-            required=True,
+            required=False,
             description="The earliest record date to sync",
         ),
+        th.Property(
+            "properties",
+            th.ArrayType(th.StringType),
+            required=True,
```
Comment (nidhi-akkio) [inline tap_hubspot/tap.py:121]: This is listed as required. But the comment mentions that we should get all if Comma-separated list of properties is not present. 

If properties are required, would we ever "get all"?

File: tap_hubspot/tap.py [hunk 2]
```diff
@@ -98,7 +98,7 @@ class TapHubspot(Tap):
             description="OAuth client id",
         ),
         th.Property(
-            "client_id",
+            "client_secret",
```
Comment (anden-akkio) [inline tap_hubspot/tap.py:101]: I think there's a way to flag to the constructor that this is a "secret" value that should be masked in the Meltano logs

See `secret` here: https://sdk.meltano.com/en/latest/classes/typing/singer_sdk.typing.Property.html#singer_sdk.typing.Property

**Review by anden-akkio — APPROVED** [review]
lgtm

File: tap_hubspot/tap.py (hunk 2 above)
Comment (matt-akkio) [reply tap_hubspot/tap.py:101]: Good catch

File: tap_hubspot/tap.py (hunk 1 above)
Comment (matt-akkio) [reply tap_hubspot/tap.py:121]: Yeah, I think you're right. That's a property I added so I'll edit

**Pull request merged**